nucleus/deploy/client.py

```diff
@@ -198,6 +210,8 @@ def create_model_endpoint(
         min_workers: int,
         max_workers: int,
         per_worker: int,
+        aws_role: str,
```

Comment:
Don't think we should be exposing `aws_role` and `results_s3_bucket`, Yi talked about it at https://github.com/scaleapi/nucleus-python-client/pull/218#discussion_r809596648

There is functionality to add in aws_role and results_s3_bucket, by calling `client.register_endpoint_auth_decorator_fn` with a function 
```
def endpoint_auth_fn(payload):
    payload['aws_role'] = "<aws_role>"
    payload["results_s3_bucket"] = "<results_bucket>"
    return payload
```

Reply:
Yeah are you working off of an older branch?